### DISCUSSION TITLE
Error during installation

### DISCUSSION THREAD

**Discussion by almasrient** [saade/filament-fullcalendar#142] [Q&A] [answered]
![image](https://github.com/saade/filament-fullcalendar/assets/140850598/69fbd9d6-17af-446a-82bc-54ee72c790d8)

1. Follow steps from here, but the widget page not created.

2. Then follow steps from below URL, then above error appears.
https://laraveldaily.com/post/filament-show-calendar-of-tasks-with-fullcalendar

**Comment by sinnbeck** [comment]
Did you run this?

>php artisan make:filament-widget CalendarWidget

It is the command that makes the file

**Reply by almasrient** [reply]
Yes, I did. And here is the files generated with both steps.

![Screenshot 2023-12-21 151950](https://github.com/saade/filament-fullcalendar/assets/140850598/04fdc2de-0f97-48ca-8b5c-d800b004ddcb)

**Reply by sinnbeck** [reply]
It looks to me like it's in the Livewire folder not Filament/Widgets?

**Reply by almasrient** [reply]
I am also not sure why.

**Reply by almasrient** [reply]
Here is the source code:
https://github.com/almasrient/MMS-F

**Reply by sinnbeck** [reply, marked as answer]
Move the file to the correct folder.

**Reply by almasrient** [reply]
I rename the Livewire folder into Widgets and move into Filament folder.
There is a Calendar menu shows and the Calendar pages appear.

Now, the Task button having a problem.
I did create the model with below fields:

id
Title
Descriptions
Start
End
timestamp

Try to understand what is the requirement to this error message.

![image](https://github.com/saade/filament-fullcalendar/assets/140850598/ab362f8b-d2e6-47fc-b7a7-6d296a41aff7)

**Reply by almasrient** [reply]
The app\Filament\Widgets\CalendarWidget.php 

![image](https://github.com/saade/filament-fullcalendar/assets/140850598/a67c4d67-1845-4b55-b0c9-1fea4a05a0cb)

**Comment by almasrient** [comment]
Now, able to create the Task/Event and data has been able to be inserted into the database.
However, unable to view/show the Task/Event Title on the calendar.
Do I miss anything here is my code: 

`<?php

namespace App\Filament\Widgets;
 
use App\Models\Task;
use Illuminate\Database\Eloquent\Model;
use Saade\FilamentFullCalendar\Widgets\FullCalendarWidget;
use Saade\FilamentFullCalendar\Actions;
// use Saade\FilamentFullCalendar\Data\EventData;
use Filament\Forms;
use Filament\Forms\Form;
use Filament\Tables;
use Filament\Tables\Table;
 
class CalendarWidget extends FullCalendarWidget
{
    public Model | string | null $model = Task::class;
 

    public function fetchEvents(array $fetchInfo): array
    {
        return Task::query()
            ->where('Start', '>=', $fetchInfo['start'])
            ->where('End', '<=', $fetchInfo['end'])
            ->get()
            ->map(
                fn (Task $event) => [
                    'Title' => $event->title,
                    'Descriptions' => $event->desc,
                    'Start' => $event->start,
                    'End' => $event->end,
                ]
            )
            ->all();
    }

    protected function headerActions(): array
    {
        return [
            Actions\CreateAction::make(),
        ];
    }

    protected function modalActions(): array
    {
        return [
            Actions\EditAction::make(),
            Actions\DeleteAction::make(),
        ];
    }

    // protected function viewAction(): Action
    // {
    //     return Actions\ViewAction::make();
    // }    
   
    public function getFormSchema(): array
    {
        return [
            Forms\Components\TextInput::make('Title'),
            Forms\Components\Textarea::make('Descriptions'),
            Forms\Components\Grid::make()
                ->schema([
                    Forms\Components\DateTimePicker::make('Start'),
                    Forms\Components\DateTimePicker::make('Ends'),
                ]),
        ];
    }

 
    public static function canView(): bool
    {
        return false;
    }
}`

![image](https://github.com/saade/filament-fullcalendar/assets/140850598/18bfbf2d-d1ad-4c1b-aa40-d857c58feca1)
